lib/index.js

```diff
@@ -354,11 +354,11 @@ Segment.prototype.retrieveCrossDomainId = function(callback) {
     var writeKey = this.options.apiKey;
 
     // Exclude the current domain from the list of servers we're querying
-    var currentTld = getTld(window.location.hostname);
+    var currentTld = topDomain(window.location.hostname);
     var domains = [];
     for (var i=0; i<this.options.crossDomainIdServers.length; i++) {
       var domain = this.options.crossDomainIdServers[i];
-      if (getTld(domain) !== currentTld) {
```

Comment:
Wait actually it does work due to the fact that you will be able to get the top domain of the xid server that corresponds to current domain.. I wonder if we can add a test case for this at all though.